experanto/data.py

```diff
@@ -5,22 +5,21 @@
 from .experiment import Experiment
 
 
-class Mouse2pDataset(Dataset):
+class Mouse2pChunkedDataset(Dataset):
     def __init__(self, root_folder: str, sampling_rate: float, chunk_size: int) -> None:
         self.root_folder = Path(root_folder)
         self.sampling_rate = sampling_rate
         self.chunk_size = chunk_size
-        self._experiment = Experiment(root_folder, sampling_rate)
-        self._first_stim_sample, self._last_stim_sample = self._experiment.get_valid_range("screen")
+        self._experiment = Experiment(root_folder)
+        self.device_names = self._experiment.device_names
+        self.start_time, self.end_time = self._experiment.get_valid_range("screen")
+        self._sample_times = np.arange(self.start_time, self.end_time, 1.0 / self.sampling_rate)
 
     def __len__(self):
-        return int((self._last_stim_sample - self._first_stim_sample) / self.chunk_size)
+        return int(len(self._sample_times) / self.chunk_size)
 
     def __getitem__(self, idx):
         s = idx * self.chunk_size
-        chunk_idx = slice(s, s + self.chunk_size)
-        return list(self._experiment[chunk_idx].values())
-
-    def get_device_names(self):
-        return self._experiment.device_names
-    
\ No newline at end of file
+        times = self._sample_times[s:s+self.chunk_size]
+        data, _ = self._experiment.interpolate(times)
+        return list(data.values())
```

Comment:
does that ensure that we always get the values in the correct order? I think it might be best to transition to `namedtuple` for return values.